diff --git a/main.qml b/main.qml
index d769954..ae2fe13 100644
--- a/main.qml
+++ b/main.qml
@@ -102,7 +102,7 @@ Window {
             /*  Keyboard visible */
 			webEngineView.height = webEngineView.height-Qt.inputMethod.keyboardRectangle.height
             /*Triggers timer to emulate a keypress after keyboard is fully shown */
-            keypr.running = true
+            //	keypr.running = true
 		}else{
             /*  Keyboard invisble  */
 			webEngineView.height = webEngineView.height+Qt.inputMethod.keyboardRectangle.height
